core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -672,6 +672,22 @@ private[spark] class ExternalSorter[K, V, C](
     partitionedIterator.flatMap(pair => pair._2)
   }
 
+  /**
+   * Return an iterator over all the data written to this object, aggregated by our aggregator.
+   * On task completion (success, failure, or cancellation), it updates related task metrics,
```

Comment:
IIUC, we only do `stop()` on task completion. Task metrics updates happen once we call `completionIterator`. Could you reword it more accurately?